packages/shared/blockchain/evm-based/src/lib/transactions/smartContract/index.ts

```diff
@@ -18,7 +18,7 @@ const smartContractWriteMethodInvocation = async (
   const contract = new client.eth.Contract([methodABI])
 
   const tx: TransactionConfig = {
-    from: 0,
+    from: undefined,
```

Comment:
from should be always 0 - why are you putting undefined everywhere? from:0 means that we are using index 0 of eth,.accounts.wallet keys, which are populated from the private Key - take a look how KLAY is done

Reply:
in all other chains we used 'undefined' because it was throwing errors while testing when it was 0. I don't know how to solve that

Reply:
In some cases of ERC20 I got:

> Provided address 0 is invalid, the capitalization checksum test failed, or it’s an indirect IBAN address which can’t be converted.

for 0 from value.

Reply:
because you dont have set the default wallet account - in case you do so - and it is desired - you wont receive it

Reply:
@jakcinmarina pls lets try it with '0'.

Reply:
'0' didn't work, 0 doesn't work as well. 
Are we setting the default address wrongly? @samuelsramko